Source/WebKit/UIProcess/Automation/BidiStorageAgent.cpp

```diff
@@ -0,0 +1,302 @@
+                matchingCookies.append(cookie);
+        }
+
+        callback({ { buildArrayForCookies(matchingCookies), partitionKey } });
+    });
+
+}
+
+void BidiStorageAgent::setCookie(Ref<JSON::Object>&& cookie, RefPtr<JSON::Object>&& optionalPartition, Inspector::CommandCallback<Ref<PartitionKey>>&& callback)
+{
+    RefPtr session = m_session.get();
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!session, InternalError);
+
+    auto parsedPartitionKey = makePartitionKey(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedPartitionKey, InternalError);
+    auto partitionKey = parsedPartitionKey.value();
+
+    auto parsedCookieStore = cookieStoreForPartition(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedCookieStore, InternalError);
+    auto cookieStore = parsedCookieStore.value();
+
+    String name = cookie->getString("name"_s);
+    String value = cookie->getString("value"_s);
+    String domain = cookie->getString("domain"_s);
+
+    if (name.isEmpty() || value.isEmpty() || domain.isEmpty()) {
+        LOG(Automation, "setCookie: Missing required fields (name, value, domain)");
+        return;
+    }
+
+    WebCore::Cookie webCoreCookie;
+    webCoreCookie.name = name;
+    webCoreCookie.value = value;
+    webCoreCookie.domain = domain;
+    webCoreCookie.path = cookie->getString("path"_s);
+
+    auto secureOptional = cookie->getBoolean("secure"_s);
+    webCoreCookie.secure = secureOptional.value_or(false);
+
+    auto httpOnlyOptional = cookie->getBoolean("httpOnly"_s);
+    webCoreCookie.httpOnly = httpOnlyOptional.value_or(false);
+
+    Vector<WebCore::Cookie> cookiesToSet;
+    cookiesToSet.append(WTFMove(webCoreCookie));
+
+    cookieStore->setCookies(WTFMove(cookiesToSet), [callback = WTFMove(callback), partitionKey = WTFMove(partitionKey)]() mutable {
+        callback({ WTFMove(partitionKey) });
+    });
+};
+
+static void deleteCookiesSequentially(API::HTTPCookieStore* store, Vector<WebCore::Cookie> cookies, size_t index, Ref<PartitionKey> partitionKey, Inspector::CommandCallback<Ref<PartitionKey>> callback)
+{
+    if (index >= cookies.size()) {
+        LOG(Automation, "deleteCookies all deletions complete.");
+        callback({ partitionKey });
+        return;
+    }
+    store->deleteCookie(cookies[index], [store, cookies = WTFMove(cookies), index, partitionKey = WTFMove(partitionKey), callback = WTFMove(callback)]() mutable {
+        deleteCookiesSequentially(store, WTFMove(cookies), index + 1, partitionKey, WTFMove(callback));
+    });
+}
+
+void BidiStorageAgent::deleteCookies(RefPtr<JSON::Object>&& optionalFilter, RefPtr<JSON::Object>&& optionalPartition, Inspector::CommandCallback<Ref<PartitionKey>>&& callback)
+{
+
+    RefPtr session = m_session.get();
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!session, InternalError);
+
+    auto parsedPartitionKey = makePartitionKey(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedPartitionKey, InternalError);
```

Comment:
Won't this error out if a filter is passed but not a partition key?

Based on my reading of the spec for this endpoint, filters are applied after obtaining the cookie store from the partition key (using a default partition key if not provided). If there are no filters it should still work.